### PR TITLE
Remove unreachable code

### PR DESCRIPTION
**Pull request by mdeweerd**
Replacement PR for #41.
Remove code that was not covered and that is functionnally unreachable. .

### CODE REVIEW

File: pre_commit_hooks/insert_license.py [hunk 1]
```diff
@@ -168,18 +168,24 @@ def process_files(args, changed_files, todo_files, license_info):
...
+    # returns except for errors that are not a "UnicodeDecodeError".
+    # Note: The chardet library could be used to support more encodings
+
+    for enc in ('utf8', 'ISO-8859-1'):
         try:
-            with open(src_filepath, encoding=encoding) as src_file:
-                return src_file.readlines(), encoding
-        except UnicodeDecodeError as error:
-            last_error = error
-    print(f"Error while processing: {src_filepath} - file encoding is probably not supported")
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:179]: I'm going to repeat my question from #41:
I found this error message quite helpful!
Why do you want to get rid of it?

Comment (mdeweerd) [reply pre_commit_hooks/insert_license.py:179]: The error is unreachable.  I provided sample code to demonstrate that.

ISO-8859-1 does not fail in that example.  The error message possibly helped before you had ISO-8859-1.

``python
#!/bin/env python3
src_filepath="allchars.txt"
f=open(src_filepath,"wb")

arr=bytearray(list(range(256)))
f.write(arr)
f.close()

for encoding in ('utf8', 'ISO-8859-1'):
    try:
        with open(src_filepath,encoding=encoding) as src_file:
            contents = src_file.readlines()
            print(f"All good with {encoding}\n")
            break
    except UnicodeDecodeError:
        print(f"UnicodeDecodeEror with {encoding}\n")
```

The output on my end is:
```
UnicodeDecodeEror with utf8

All good with ISO-8859-1
```

Comment (Lucas-C) [reply pre_commit_hooks/insert_license.py:179]: I totally agree with you, but the error is here for ALL other encodings that utf8 & ISO-8859-1 !! 😊

Comment (mdeweerd) [reply pre_commit_hooks/insert_license.py:179]: The code first tests UTF8, then ISO-8859-1 and will never fail on the latter, so the original code always returns inside the loop.  The code after the loop is never executed.

**Pull request closed without merging**